src/Illuminate/Support/Collection.php

```diff
@@ -768,6 +768,16 @@ public function toJson($options = 0)
 	}
 
 	/**
+	 * Get an hash of the collection.
+	 *
+	 * @return string
+	 */
+	public function hash()
+	{
+		return md5($this);
```

Comment:
[SHA-1 is better than MD5 when we are speaking in terms of collision-resistance.](http://security.stackexchange.com/questions/19705/is-sha1-better-than-md5-only-because-it-generates-a-hash-of-160-bits)